File: src/main/java/DayTwoChecksum.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class DayTwoChecksum {

    int calculateCheckSum(List<String> rows) {
        return rows.stream().mapToInt(this::calculateRowChecksum).sum();
    }

    int calculateRowChecksum(String row) {
        String[] numbers = row.split("\\W");
        int max = Stream.of(numbers).mapToInt(Integer::parseInt).max().orElse(0);
        int min = Stream.of(numbers).mapToInt(Integer::parseInt).min().orElse(0);
        return max - min;
    }

    public int calculateFactorChecksum(String row) {
        String[] numbers = row.split("\\W");
        List<Integer> integers = Stream.of(numbers).mapToInt(Integer::parseInt).boxed().collect(Collectors.toList());

        for (int i = 0; i < integers.size() - 1; i++) {
            for (int j = i + 1; j < integers.size(); j++) {
                Integer numberone = integers.get(i);
                Integer numbertwo = integers.get(j);
                if (numberone % numbertwo == 0) {
                    return numberone / numbertwo;
                }
                if (numbertwo % numberone == 0) {
                    return numbertwo / numberone;
                }
            }
        }
        return 0;
    }
}

File: src/main/java/DayTwoChecksum.kt
import kotlin.streams.toList

class DayTwoChecksum {

    internal fun calculateDiffChecksum(rows: List<String>): Int {
        return rows.stream().mapToInt { this.calculateRowDiffChecksum(it) }.sum()
    }

    internal fun calculateRowDiffChecksum(row: String): Int {
        val numbers = row.split(Regex("\\W"))
        val max = numbers.stream().mapToInt { Integer.parseInt(it) }.max().orElse(0)
        val min = numbers.stream().mapToInt { Integer.parseInt(it) }.min().orElse(0)
        return max - min
    }

    fun calculateFactorRowChecksum(row: String): Int {
        val integers = row.split(Regex("\\W")).stream()
                .mapToInt { Integer.parseInt(it) }
                .boxed()
                .toList()

        for (i in 0 until integers.size - 1) {
            for (j in i + 1 until integers.size) {
                val numberone = integers[i]
                val numbertwo = integers[j]
                if (numberone % numbertwo == 0) {
                    return numberone / numbertwo
                }
                if (numbertwo % numberone == 0) {
                    return numbertwo / numberone
                }
            }
        }
        return 0
    }

    fun calculateFactorCheckSum(rows: List<String>): Int {
        return rows.stream().mapToInt { this.calculateFactorRowChecksum(it) }.sum()
    }
}
